Add text regarding including shared analysis options (#6402)
dart-lang/site-www: src/content/tools/analysis.md

or [disable individual rules][].
:::

:::note

> For more information about including options files, see [including shared
> options](#including-shared-options).

Reviewer: ```suggestion
For more information about including options files, 
check out the [Including shared options](#including-shared-options) section.
```

You can use the other syntax for rules in an included file.
:::

## Including shared options

> An analysis options file can include other options which are specified in

Reviewer: ```suggestion
An analysis options file can include options which are specified in
```

> another options file, or even a list of other options files. Specify such files
> with the top-level `include:` field:

Reviewer: ```suggestion
another options file, or even a list of other options files.
You can specify such files using the top-level `include:` field:
```

```yaml title="analysis_options.yaml"
include: package:flutter_lints/recommended.yaml
```

An included options file can be specified with a `package:` path, or a relative
path. Multiple analysis options files can be specified in a list:

```yaml title="analysis_options.yaml"
include:
  - package:flutter_lints/recommended.yaml
  - ../team_options.yaml
```

> Options found in an included file can be overridden in the including file.
> Options found in an included file can also be overridden by subsequent included
> files. In other words, the options specified by an analysis options file are

Reviewer: ```suggestion
Options in an included file can be overridden in the including file,
as well as by subsequent included files. 
In other words, the options specified by an analysis options file are
```

computed by first applying the options specified in each of the included files
(by recursively applying this algorithm), in the order they appear in the list,
and then overriding them with any locally defined options.

For example, given the following options files:

```yaml title="three.yaml"
include: two.yaml
# ...
```

> and a final options file that includes these:

Reviewer: ```suggestion
And a final options file that includes these:
```

```yaml title="analysis_options.yaml"
include:
  - one.yaml
  - three.yaml
# ...
```

> then the combined analysis options are computed by applying the options found

Reviewer: ```suggestion
Then the combined analysis options are computed by applying the options found
```

You can use the other syntax for rules in an included file.
:::

## Including shared options

> An analysis options file can include other options which are specified another

Reviewer 2: "specified another" --> "specified in another"

Author: Done

options file, or even a list of other options files. Specify such files with
the top-level `include:` field:

```yaml title="analysis_options.yaml"
include: package:flutter_lints/recommended.yaml
```

An included options file can be specified with a `package:` path, or a relative

> path. Multiple analysis options can be specified in a list:

Reviewer 2: "options" --> "option files"

Author: Done

```yaml title="analysis_options.yaml"
include:
  - package:flutter_lints/recommended.yaml
  - ../team_options.yaml
```

Options found in an included file can be overridden in the including file.
Options found in an included file can also be overridden by subsequent included

> files. For example, given the following options files:

Reviewer 2: I think it would be helpful to have a brief textual description of the algorithm, perhaps something like
```
files. In other words, the options specified by an analysis options file are computed by first applying the options specified in each of the included files (by recursively applying this algorithm), in the order they appear in the list, and then overriding them with any locally defined options.

For example, given the following options files:
```

I'll point out that there's no example demonstrating the ability of a file later in the list to override files earlier in the list, and that might be useful.

I'm not most readers, so this might not apply, but I'd actually find the examples to be easier to follow if they were more abstract. For me, I'd rather just see a description of the form: The options in the files will be computed by first applying `one.yaml`, then `two.yaml`, then `three.yaml`, and finally `analysis_options.yaml`.

Author: Thank you for both of these suggestions! I actually could not figure out how I would write the algorithm succinctly, and gave up. But the sentence you have here is perfect.

Also I like describing the application order, instead of trying to map back what file specified what option. I've simplified the example to spell out what you wrote.
